### ISSUE TITLE
How to add buttons on the header?

### ISSUE THREAD

**Issue by xinnai** [measuredco/puck#348]
Hi, your lib is amazing, thank you! Now I have some problem:
1、I want to add a back button on the left-side of the header. But  when I override the **header**, the whole header is gone, when I override the **headerActions**, the button will always on the right, how to just add a left button?
2、I want to modify the css of the buttons on the header, like `undo`,`redo`,`publish` button, but I don't know how. I tried to add a new publish button by override the **headerActions**, but the default publish button is still there and I don' know how to hide it. If I add new `undo`,`redo` buttons insdead of modify the css, I just don't know how to invoke the undo and redo functions in Puck.
Can you help?

**Comment by chrisvxd** [comment]
Thanks @xinnai!

1. You can't do that without overriding the entire **header**, but you can reimplement all the buttons using the `usePuck` hook.
2. There's no way to modify the CSS at the moment, but we should have a solution once we tackle #139

**Comment by chrisvxd** [comment]
There's an example in the custom UI example:

```

const CustomHeader = ({ onPublish }: { onPublish: (data: Data) => void }) => {
  const { appState, dispatch } = usePuck();

  return (
    <header>
      <span style={{ fontWeight: 600 }}>Custom UI example </span>
      <div style={{ marginLeft: "auto", display: "flex", gap: 8 }}>
        <div>
          <Button
            onClick={() => onPublish(appState.data)}
            icon={<Globe size="14" />}
          >
            Publish
          </Button>
        </div>
      </div>
    </header>
  );
};
```

To implement the history buttons, you'll need to use `history` from `usePuck`, implemented in #342 but not yet released or documented. If you need to implement that, you'll need to use the latest canary build (see [npm](https://www.npmjs.com/package/@measured/puck?activeTab=versions).
